plugins/modules/group.py

```diff
@@ -567,11 +567,12 @@ def perform_module_operation(self):
                                   users_list)
                 changed = True
             else:
-                id_exists = self.check_if_id_exists(group_name, group_details)
-                if id_exists and group_name is not None:
-                    error_message = f'Group already exists with GID {group_id}'
-                    LOG.error(error_message)
-                    self.module.fail_json(msg=error_message)
+                if group_id and group_name:
+                    id_exists = self.check_if_id_exists(group_name, group_details)
+                    if id_exists and group_name is not None:
```

Comment:
Do we need to check for group_name again since we are already checking in the parent condition?